corehq/apps/app_manager/static/app_manager/js/releases.js

```diff
@@ -7,13 +7,49 @@ function SavedApp(o, r) {
     if (!self.include_media) {
         self.include_media = ko.observable(false);
     }
+    if(!self.generating_url){
+        self.generating_url = ko.observable(false);
+    }
+
+    self.generate_short_url = function(url_type){
+        //accepted_url_types = ['short_odk_url', 'short_odk_media_url', 'short_url']
+        if (!url_type){ url_type = 'short_odk_url'; }
+        var base_url = '/a/' + self.domain() + '/apps/odk/' + self.id() + '/',
+            should_generate_url = ((url_type === 'short_odk_url') && self.short_odk_url && !self.short_odk_url()) ||
+                                  ((url_type === 'short_odk_media_url') && self.short_odk_media_url && !self.short_odk_media_url()) ||
+                                  ((url_type === 'short_url') && self.short_url && !self.short_url());
+        
+        if (should_generate_url && !self.generating_url()){
+            self.generating_url(true);
+            $.ajax({
+                url: base_url + url_type + '/'
```

Comment:
Yeah, still works without the `'/'`, but without it, the server issues a `301`, before serving the content I want.